duckduckgo/Android
# FireproofRepository API injection

Task/Issue URL:https://app.asana.com/0/0/1203433965574189/f

### Description
Update all injection references of final class FireproofRepository to FireproofRepository interface

### Steps to test this PR

- Install from this branch
- Check fireproof feature works as expected
          - Fireproof website
          - Fireproof settings section
          - Tap fire button and check fireproof websites are still saved

### No UI changes
